src/data/datasets/representation/validation.py

```diff
@@ -0,0 +1,288 @@
+    Where gallery items are treated non-relevant.
+
+    Example matches csv:
+    Query ids should be unique, otherwise the rows having the same query id will be treated as different matches.
+    Relevant ids can be repeated in different queries.
+    Scores reflect the order of similarity of the image to the query,
+    a higher score corresponds to a greater similarity.
+
+    .. csv-table:: Match csv example
+    :header: query, relevant, scores
+    1194917,601566 554492 224125 2001716519,4 3 2 2
+    1257924,456490,4
+
+    Example img_list csv:
+    img_list.csv maps the id's of query and relevant elements to image paths
+
+    .. csv-table:: Image csv example
+    :header: id, image_path
+    1194917,data/img_1.jpg
+    601566,data/img_2.jpg
+    554492,data/img_3.jpg
+    224125,data/img_4.jpg
+    2001716519,data/img_5.jpg
+    1257924,data/img_6.jpg
+    456490,data/img_7.jpg
+
+    .. csv-table:: Gallery Image csv example
+    :header: id, image_paths
+    8,data/db/img_1.jpg
+    10,data/db/img_2.jpg
+    12,data/db/img_3.jpg
+    """
+
+    def __init__(self,
+                 data_folder: str,
+                 matches_csv_path: str,
+                 img_list_csv_path: str,
+                 transform: Union[BasicTransform, BaseCompose],
+                 augment: Optional[Union[BasicTransform, BaseCompose]] = None,
+                 use_gallery: Optional[bool] = False,
+                 gallery_folder: Optional[str] = None,
+                 gallery_list_csv_path: Optional[str] = None,
+                 image_dtype: str = 'float32',
+                 img_list_map_column: dict = {'image_path':'image_path',
+                                              'img_id': 'id'},
+                 matches_map_column: dict = {'query': 'query', 
+                                             'relevant':'relevant', 
+                                             'scores': 'scores'},
+
+                 gallery_map_column: dict = {'gallery_path':'image_path',
+                                             'gallery_id':'id'},
+                 grayscale: bool = False):
+        """
+        Args:
+            data_folder: Directory with all the images.
+            matches_csv_path: path to csv file where queries with their relevance scores are specified
+            img_list_csv_path: path to mapping image identifiers to image paths. Format: id | path.
+                Id from matches csv are linked to id from img_list csv
+            transform: Transform to be applied on a sample. This should have the
+                interface of transforms in `albumentations` library.
+            augment: Optional augment to be applied on a sample.
+                This should have the interface of transforms in `albumentations` library.
+            use_gallery: If True, will use gallery data(non-relevant items)
+            gallery_folder: path to a folder with all gallery images (traversed recursively)
```

Comment:
Add also: `When the gallery not specified all the remaining queries and relevant will be considered as negative samples to a given query-relevant set`